File: swift/Sources/LibSignalClient/zkgroup/CallLinkPublicParams.swift
//
// Copyright 2023 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class CallLinkPublicParams: ByteArray, @unchecked Sendable {
    public required init(contents: [UInt8]) throws {
        try super.init(contents, checkValid: signal_call_link_public_params_check_valid_contents)
    }
}

File: node/ts/zkgroup/calllinks/CallLinkPublicParams.ts
//
// Copyright 2023 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray.js';
import * as Native from '../../Native.js';

export default class CallLinkPublicParams extends ByteArray {
  private readonly __type?: never;

  constructor(contents: Uint8Array) {
    super(contents, Native.CallLinkPublicParams_CheckValidContents);
  }
}
